src/bytes/bytes_lemmas.gobra

```diff
@@ -0,0 +1,359 @@
+package bytes
+
+// +gobra
+
+import (
+	. "verification/utils/definitions"
+	// b "verification/utils/bitwise"
+	sl "gobra-libs/byteslice"
+	bytes "gobra-libs/bytes"
+	// seqs "verification/utils/seqs"
+	// sets "verification/utils/sets"
+	. "bytes/spec"
+)
+
+ghost
+requires 0 < p
+requires acc(sl.Bytes(a, 0, len(a)), p)
+requires acc(sl.Bytes(b, 0, len(b)), p)
+requires r == ((unfolding acc(sl.Bytes(a, 0, len(a)), p) in string(a)) == (unfolding acc(sl.Bytes(b, 0, len(b)), p) in string(b)))
```

Comment:
Whenever possible, we should try to avoid unfoldings in specs. They make everything harder to read. I would suggest introducing a pure function `String(s []byte)` whose body is the unfolding you have here.
```suggestion
requires r == (String(a) == String(b))
```

Reply:
Also, please mark this as trusted and add a comment explaining that we cannot currently prove this due to our incomplete encoding of strings